doc/api/http.md

```diff
@@ -1210,7 +1221,8 @@ added: v0.4.0
 
 Sets a single header value for implicit headers. If this header already exists
 in the to-be-sent headers, its value will be replaced. Use an array of strings
-here to send multiple headers with the same name.
+here to send multiple headers with the same name. Non-string values will be
+coerced to strings.
```

Comment:
Non-string values will only be coerced to strings before these data will be transmitted.

Reply:
They are coerced to strings already during set in `validateHeader()` => `checkInvalidHeaderChar()` but the result is not stored. if it fails or holds invalid chars caller gets an Error.

Reply:
Ooof, I see the subtlety here now. Maybe be super explicit?

Non-string values will be stored without modification. Therefore, `getHeader()` may return non-string values. However, the non-string values will be converted to strings for network transmission.

Something like that maybe?